beacon-chain/operations/attestations/kv/block.go

```diff
@@ -48,10 +48,10 @@ func (p *AttCaches) SaveBlockAttestations(atts []*ethpb.Attestation) error {
 
 // BlockAttestations returns the block attestations in cache.
 func (p *AttCaches) BlockAttestations() []*ethpb.Attestation {
-	atts := make([]*ethpb.Attestation, 0)
-
 	p.blockAttLock.RLock()
 	defer p.blockAttLock.RUnlock()
+
+	atts := make([]*ethpb.Attestation, 0, len(p.blockAtt))
```

Comment:
This doesn't look safe, looks like p.blockAtt is a double array?

Reply:
which part isnt safe ? this just adds in the capacity of the slice

Reply:
Oh shoot, I think @0xKiwi has a good point. This p.blockAtt is a map of arrays. Let me double check this is correct...

Reply:
Yeah.. I'm going to revert this one. It's not unsafe but it's also not correct at all.

We might have a map like 
0x01 -> [att, att, att]
0x02 -> [att, att, att]
0xAB -> [att, att, att]

The code here would create and only allocate enough memory for 3 attestations when the true number was 9.